File: projetoJava/src/Configuracao.java
import java.util.HashMap;

public class Configuracao {

    private static Configuracao instancia = null;

    private HashMap<String, String> configuracoes;

    private Configuracao() {
        configuracoes = new HashMap<>();
    }

    public static Configuracao getInstancia() {
        if (instancia == null) {
            instancia = new Configuracao();
        }
        return instancia;
    }

    public void setConfiguracao(String chave, String valor) {
        configuracoes.put(chave, valor);
    }

    public String getConfiguracao(String chave) {
        return configuracoes.getOrDefault(chave, null);
    }
}

File: projetoGo/configuracao.go
package main

import "sync"

type Configuracao struct {
	configuracoes map[string]string
}

var instancia *Configuracao

var mu sync.Mutex

func GetConfiguracao() *Configuracao {
	mu.Lock()
	defer mu.Unlock()

	if instancia == nil {
		instancia = &Configuracao{
			configuracoes: make(map[string]string),
		}
	}
	return instancia
}

func (c *Configuracao) SetConfiguracao(chave, valor string) {
	c.configuracoes[chave] = valor
}

func (c *Configuracao) GetConfiguracao(chave string) string {
	return c.configuracoes[chave]
}
